statsmodels/base/data.py

```diff
@@ -190,47 +191,43 @@ def handle_missing(cls, endog, exog, missing, **kwargs):
         kwargs. It preserves Nones.
         """
         none_array_names = []
+        combined_names = []
+        combined = []
 
         # patsy's already dropped NaNs in y/X
         missing_idx = kwargs.pop('missing_idx', None)
 
         if missing_idx is not None:
             # y, X already handled by patsy. add back in later.
-            combined = ()
-            combined_names = []
             if exog is None:
-                none_array_names += ['exog']
+                none_array_names.append('exog')
         elif exog is not None:
-            combined = (endog, exog)
-            combined_names = ['endog', 'exog']
+            combined.extend([endog, exog])
+            combined_names.extend(['endog', 'exog'])
         else:
-            combined = (endog,)
-            combined_names = ['endog']
-            none_array_names += ['exog']
+            combined.append(endog)
+            combined_names.append('endog')
+            none_array_names.append('exog')
```

Comment:
This can be further simplified, but I'm saving that for a follow-up to keep the diff easily-checked.